staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1291,12 +1299,27 @@ func (c *Cacher) getCommonResourceVersionLockedFunc(ctx context.Context, parsedW
 	}
 }
 
+// getWatchCacheResourceVersion returns a ResourceVersion to which the watch cache must be synchronized to
+//
+// Depending on the input parameters, the semantics of the returned ResourceVersion are:
+//   - must be at Exact RV (when parsedWatchResourceVersion > 0)
+//   - can be at Any RV (when parsedWatchResourceVersion = 0)
+//   - must be at Most Recent RV (return an RV from etcd)
+func (c *Cacher) getWatchCacheResourceVersion(parsedWatchResourceVersion uint64, opts storage.ListOptions, getCurrentResourceVersionFromStorageOnce func() (uint64, error)) (uint64, error) {
```

Comment:
locked?

Reply:
no, we don't need a lock here. We would need one if we were to read the current `RV` from the `watchCache` (`Any` semantic).